src/content/app/genome-browser/hooks/useGenomeBrowserIds.ts

```diff
@@ -41,57 +31,29 @@ import {
 const useGenomeBrowserIds = () => {
   const activeGenomeId = useAppSelector(getBrowserActiveGenomeId);
   const activeFocusObjectId = useAppSelector(getBrowserActiveFocusObjectId);
-  const committedSpecies = useAppSelector((state) =>
-    getCommittedSpeciesById(state, activeGenomeId ?? '')
-  );
-  const params = useUrlParams<'genomeId'>('/genome-browser/:genomeId');
-  const { search } = useLocation();
+  const genomeBrowserIdsContext = useContext(GenomeBrowserIdsContext);
 
-  const { genomeId: genomeIdInUrl } = params;
-  const urlSearchParams = new URLSearchParams(search);
-  const focusObjectIdInUrl = urlSearchParams.get('focus');
+  if (!genomeBrowserIdsContext) {
+    throw new Error(
+      'useGenomeBrowserIds must be used with GenomeBrowserIdsContext Provider'
+    );
+  }
 
   const {
-    currentData: genomeInfo,
-    isFetching,
-    isError,
-    error
-  } = useGenomeInfoQuery(genomeIdInUrl ?? '', {
-    skip: !genomeIdInUrl
-  });
-  const genomeId = genomeInfo?.genomeId;
-  const isMissingGenomeId = isError && isGenomeNotFoundError(error);
-
-  // TODO: check if the logic below is correct
-  const genomeIdForUrl =
-    genomeIdInUrl ??
-    committedSpecies?.genome_tag ??
-    committedSpecies?.genome_id;
-
-  let focusObjectId;
-  let focusObjectIdForUrl;
-  let parsedFocusObjectId;
-  let isMalformedFocusObjectId = false;
-
-  if (focusObjectIdInUrl) {
-    focusObjectIdForUrl = focusObjectIdInUrl;
-    if (genomeId) {
-      try {
-        parsedFocusObjectId = {
-          genomeId,
-          ...parseFocusIdFromUrl(focusObjectIdInUrl)
-        };
-        focusObjectId = buildFocusObjectId(parsedFocusObjectId);
-      } catch {
-        isMalformedFocusObjectId = true;
-      }
-    }
-  } else if (activeFocusObjectId) {
-    focusObjectIdForUrl = buildFocusIdForUrl(activeFocusObjectId);
-  }
+    genomeIdInUrl,
+    genomeIdForUrl,
+    focusObjectIdInUrl,
+    isFetchingGenomeId,
+    isMissingGenomeId,
+    genomeId,
+    focusObjectId,
+    focusObjectIdForUrl,
+    parsedFocusObjectId,
+    isMalformedFocusObjectId
+  } = genomeBrowserIdsContext;
```

Comment:
Doesn't it just mean simply returning the context?

Reply:
✅